### PR TITLE
Adding handle missing values example to FreeMarker PayloadFactory

### PR DESCRIPTION
**Pull request by tharakamd**
## Purpose
Adding handle missing values example to FreeMarker PayloadFactory

## Related issues
https://github.com/wso2/micro-integrator/issues/2124
https://github.com/wso2/micro-integrator/issues/2122

### CODE REVIEW

File: en/docs/reference/mediators/payloadfactory-mediator.md [hunk 1]
```diff
@@ -1047,4 +1047,49 @@ This example shows how to use arguments in a freemarker template to pass values
     </payloadFactory>
     ```
 
-In this example, the value for the “argument one” key is replaced by the first argument value. The argument for the "argument two" key is replaced by the second argument value.
\ No newline at end of file
+In this example, the value for the “argument one” key is replaced by the first argument value. The argument for the "argument two" key is replaced by the second argument value.
+
+### Handling Missing Values
```
Comment (abeykoon) [inline en/docs/reference/mediators/payloadfactory-mediator.md:1052]: Shall we say Handling Null/Empty Values? 
The word `Missing` does not give that feeling. 

Also better mention if you know some input values can be optional in the payload, you can construct the FreeMarker template.

Comment (tharakamd) [reply en/docs/reference/mediators/payloadfactory-mediator.md]: Modified the description and the title.

**Review by abeykoon — APPROVED** [review]

File: en/docs/reference/mediators/payloadfactory-mediator.md [hunk 2]
```diff
@@ -1047,4 +1047,53 @@ This example shows how to use arguments in a freemarker template to pass values
     </payloadFactory>
     ```
 
-In this example, the value for the “argument one” key is replaced by the first argument value. The argument for the "argument two" key is replaced by the second argument value.
\ No newline at end of file
+In this example, the value for the “argument one” key is replaced by the first argument value. The argument for the "argument two" key is replaced by the second argument value.
+
+### Handling Optional Values
```
Comment (nilminiwso2) [inline en/docs/reference/mediators/payloadfactory-mediator.md:1052]:
```suggestion
### Handling optional values
```
```suggestion
### Handling Optional Values
```

File: en/docs/reference/mediators/payloadfactory-mediator.md [hunk 3]
```diff
@@ -1047,4 +1047,53 @@ This example shows how to use arguments in a freemarker template to pass values
...
+    <payloadFactory media-type="json" template-type="freemarker">
+        <format><![CDATA[{
+        "Name": "${payload.first_name} ${payload.last_name ! "" }",
+        "Age": ${payload.age}
+         }]]>
+        </format>
+    </payloadFactory>
+    ```
+
+In this example, The FreeMarker template is expecting a property named `last_name` from the input payload. But the
```
Comment (nilminiwso2) [inline en/docs/reference/mediators/payloadfactory-mediator.md:1096]:
```suggestion
In this example, The FreeMarker template is expecting a property named `last_name` from the input payload. However, the 
```
```suggestion
In this example, The FreeMarker template is expecting a property named `last_name` from the input payload. But the 
```

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**